### PR TITLE
Add API call functionality to create posting

### LINKED ISSUE #88: Admin create posting "Review and Post" : API call
## User Story and Context

As an admin (Sistering employee), I need to publish a posting so that volunteers can view it and choose to submit their availability to it.

### Product
Please see #43.

**This ticket covers the parts enclosed in the red rectangle only**:

**Figma link:** https://www.figma.com/file/TWJTNWjK75A6VgSisySTLe/Admin-Experience?node-id=186%3A2199

Please confirm with @amandatdu on which designs to use.

### Technical
This ticket is for implementing the **API call only**.
* Send posting and shifts data to backend using 2 separate requests when the user selects Post or Save as Draft

## Dev Notes

**Chakra docs for recommended components:**
* [Tag](https://chakra-ui.com/docs/data-display/tag)
* [Table](https://chakra-ui.com/docs/data-display/table)

### PR DESCRIPTION
**Pull request by briantu**
## Ticket link

Closes #88 

## Implementation description
* Added the create posting mutation in `CreatePostingReviewPage`
* Added two buttons ("Save as Draft" and "Post") in `CreatePostingReviewPage` that call the create posting mutation
  * Save as Draft: creates posting with status `DRAFT`
  * Post: creates posting with status `PUBLISHED`
* `CreatePostingReviewPage` gets posting data using `PostingContext`

## Steps to test
1. Run the application locally `docker-compose up --build`
2. Create a branch in your local database for the posting to use
3. Click on the "Edit Posting" tab on the default page and enter the branch id of the branch you just created. 
4. Add other dummy information (be sure to include the dates for "Start Date", "End Date", and "Auto-Closing Date").
5. Click on the "Review Posting" tab and click the "Save as Draft" or "Post" button
6. Check that the posting was successfully created in your local database

## What should reviewers focus on?
* Create posting functionality works as expected
* Correct coding practices for calling API

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Comment by briantu** [comment]
Question to consider - how are we handling front-end error messaging? (e.g. when API call to create posting fails)

File: frontend/src/components/pages/admin/posting/CreatePostingReviewPage.tsx [hunk 1]
```diff
@@ -1,13 +1,57 @@
...
+  }
+`;
+
 const CreatePostingReviewPage = (): React.ReactElement => {
+  const { status, times, recurrenceInterval, ...postingToCreate } = useContext(
+    PostingContext,
+  );
+  const [createPosting, { data, loading, error }] = useMutation(CREATE_POSTING);
+
+  if (loading) console.log("Submitting...");
```
Comment (LenaNguyen) [inline frontend/src/components/pages/admin/posting/CreatePostingReviewPage.tsx:25]: Remove this after adding `isLoading` prop to buttons

File: frontend/src/components/pages/admin/posting/CreatePostingReviewPage.tsx [hunk 2]
```diff
@@ -1,13 +1,57 @@
...
+  if (error) console.log(`Submission error! ${error}`);
+
   return (
     <div>
       <CreatePostingReview />
       <MainPageButton />
+      <Button
```
Comment (LenaNguyen) [inline frontend/src/components/pages/admin/posting/CreatePostingReviewPage.tsx:32]: You can add a loading state to buttons via props. See [docs](https://chakra-ui.com/docs/form/button#button-loading-state). `isLoading={loading}`

File: frontend/src/components/pages/admin/posting/CreatePostingReviewPage.tsx [hunk 3]
```diff
@@ -1,13 +1,57 @@
...
+          createPosting({
+            variables: {
+              posting: { ...postingToCreate, ...{ status: "DRAFT" } },
+            },
+          });
+        }}
+      >
+        Save as Draft
+      </Button>
+      <Button
```
Comment (LenaNguyen) [inline frontend/src/components/pages/admin/posting/CreatePostingReviewPage.tsx:44]: `isLoading={loading}`

File: frontend/src/components/pages/admin/posting/CreatePostingReviewPage.tsx [hunk 4]
```diff
@@ -1,13 +1,57 @@
...
+`;
+
 const CreatePostingReviewPage = (): React.ReactElement => {
+  const { status, times, recurrenceInterval, ...postingToCreate } = useContext(
+    PostingContext,
+  );
+  const [createPosting, { data, loading, error }] = useMutation(CREATE_POSTING);
+
+  if (loading) console.log("Submitting...");
+  if (error) console.log(`Submission error! ${error}`);
```
Comment (LenaNguyen) [inline frontend/src/components/pages/admin/posting/CreatePostingReviewPage.tsx:26]: We should get some designs for what the user should see if there's an error.

For now let's just `window.alert(error)`.

File: frontend/src/components/pages/admin/posting/CreatePostingReviewPage.tsx [hunk 5]
```diff
@@ -1,13 +1,57 @@
...
+      title
+    }
+  }
+`;
+
 const CreatePostingReviewPage = (): React.ReactElement => {
+  const { status, times, recurrenceInterval, ...postingToCreate } = useContext(
+    PostingContext,
+  );
+  const [createPosting, { data, loading, error }] = useMutation(CREATE_POSTING);
```
Comment (LenaNguyen) [inline frontend/src/components/pages/admin/posting/CreatePostingReviewPage.tsx:23]: If you're not using data, put an underscore at the beginning of the name like `_data` to ignore the lint warning

File: frontend/src/components/pages/admin/posting/CreatePostingReviewPage.tsx [hunk 6]
```diff
@@ -1,13 +1,57 @@
...
       <MainPageButton />
+      <Button
+        variant="outline"
+        onClick={() => {
+          createPosting({
+            variables: {
+              posting: { ...postingToCreate, ...{ status: "DRAFT" } },
+            },
+          });
+        }}
```
Comment (LenaNguyen) [inline frontend/src/components/pages/admin/posting/CreatePostingReviewPage.tsx:40]:
```suggestion
        onClick={() => 
          createPosting({
            variables: {
              posting: { ...postingToCreate, status: "DRAFT"  },
            },
          });
        }
```

File: frontend/src/components/pages/admin/posting/CreatePostingReviewPage.tsx [hunk 7]
```diff
@@ -1,13 +1,57 @@
...
+        Save as Draft
+      </Button>
+      <Button
+        onClick={() => {
+          createPosting({
+            variables: {
+              posting: { ...postingToCreate, ...{ status: "PUBLISHED" } },
+            },
+          });
+        }}
```
Comment (LenaNguyen) [inline frontend/src/components/pages/admin/posting/CreatePostingReviewPage.tsx:51]:
```suggestion
        onClick={() => 
          createPosting({
            variables: {
              posting: { ...postingToCreate, status: "PUBLISHED" },
            },
          });
        }
```

**Review by LenaNguyen — CHANGES_REQUESTED** [review]
sorry for the late review .-.

File: frontend/src/components/pages/admin/posting/CreatePostingReviewPage.tsx (hunk 2 above)
Comment (briantu) [reply frontend/src/components/pages/admin/posting/CreatePostingReviewPage.tsx:33]: Done! Also I created an `isDraftClicked` boolean to track which button is clicked for the loading state.

File: frontend/src/components/pages/admin/posting/CreatePostingReviewPage.tsx (hunk 5 above)
Comment (briantu) [reply frontend/src/components/pages/admin/posting/CreatePostingReviewPage.tsx]: Decided to completely remove the `data` variable

**Review by LenaNguyen — APPROVED** [review]
Nice discovery of `useBoolean`!

**Pull request merged**